androidApp/src/main/java/com/example/tasky/android/login/viewmodel/RegisterViewModel.kt

```diff
@@ -4,13 +4,19 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.example.tasky.android.login.screen.RegisterScreenEvent
 import com.example.tasky.android.login.screen.RegisterScreenState
+import com.example.tasky.model.login.RegisterBody
+import com.example.tasky.model.onError
+import com.example.tasky.model.onSuccess
+import com.example.tasky.repository.LoginRepository
 import com.example.tasky.util.Validator
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
 class RegisterViewModel : ViewModel() {
+    private val loginRepository = LoginRepository()
+
```

Comment:
I would avoid instantiating dependencies inside your `ViewModel` because it highly couples your ViewModel with this class, which makes testing difficult. Instead, what I would suggest is to create an interface for `LoginRespository` and use that. This way, your ViewModel will depend on an abstraction that you can freely swap out for a more lightweight fake implementation when you write tests - this way, you can better ensure that when a test fails, that it is because of an issue in your ViewModel.


```
class RegisterViewModel(
    private val loginRepository: LoginRepository // <— interface
) : ViewModel() { }
```